src/components/structureDefSettings/StructureDefSettings.tsx

```diff
@@ -27,71 +32,70 @@ const StructureDefSettings: React.FC<StructureDefSettingsProps> = ({
   const { complexTypes, primitiveTypes, structureDefinitionTree } = useSelector(
     (state: RootState) => state.fhirDataTypes
   );
-
   const dispatch = useDispatch();
   const classes = useStyles();
 
   const submit = () => {
     if (structureDefinition) {
       const structureDefinitonToEdit = {
-        ...structureDefMeta,
+        ...structureDefJSON,
         snapshot: { ...structureDefinition.snapshot }
       };
       if (structureDefinitionType === "resource") {
+        dispatch(setSnackbarOpen("success", "Saved !"));
         dispatch(updateStructureDefProfile(structureDefinitonToEdit));
       } else if (structureDefinitionType === "extension") {
+        dispatch(setSnackbarOpen("success", "Saved !"));
         dispatch(updateStructureDefExtension(structureDefinitonToEdit));
       }
     }
   };
 
-  const recursiveObjectAssign = (newTree: any, initialTree: any) => {
-    for (const key of Object.keys(initialTree)) {
-      if (initialTree[key] instanceof Object && newTree)
-        Object.assign(
-          initialTree[key],
-          recursiveObjectAssign(newTree[key], initialTree[key])
-        );
-    }
-    Object.assign(newTree || {}, initialTree);
-    return newTree;
-  };
-
-  const structureDefJSON = recursiveObjectAssign(
-    createJSONTree(structureDefinitionTree),
-    structureDefinition
+  const emptyTree = createJSONTree(
+    structureDefinitionTree,
+    cloneDeep(structureDefinition)
+  );
+  const [structureDefJSON, setStructureDefJSON] = useState(
+    merge(cloneDeep(emptyTree), structureDefinition)
   );
 
-  const [structureDefMeta, setStructureDefMeta] = useState(structureDefJSON);
+  const onDeleteComplexType = (path: string, i: number) => {
+    const str: IStructureDefinition = { ...structureDefJSON };
+    let structureDefJSONAttr: any = get(str, path);
+    structureDefJSONAttr.splice(i, 1);
+    setStructureDefJSON(str);
```

Comment:
Be careful, you should be aware that you are mutating structureDefJSON because you don't deepClone it before accessing the path target. In this case there is no consequences but it could lead to bugs in other cases.

Reply:
@simonvadee